crates/miden-lib/src/account/interface/mod.rs

```diff
@@ -0,0 +1,359 @@
+    pub fn is_faucet(&self) -> bool {
+        self.account_id.is_faucet()
+    }
+
+    /// Returns true if the reference account is a regular.
+    pub fn is_regular_account(&self) -> bool {
+        self.account_id.is_regular_account()
+    }
+
+    /// Returns true if the reference account is public.
+    pub fn is_public(&self) -> bool {
+        self.account_id.is_public()
+    }
+
+    /// Returns a reference to the vector of used authentication schemes.
+    pub fn auth(&self) -> &Vec<AuthScheme> {
+        &self.auth
+    }
+
+    /// Returns a reference to the set of used component interfaces.
+    pub fn interfaces(&self) -> &Vec<AccountComponentInterface> {
+        &self.interfaces
+    }
+
+    /// checks if the provided note is compatible with the current [AccountInterface].
+    pub fn is_compatible(&self, note: &Note) -> NoteAccountCompatibility {
+        if let Some(well_known_note) = WellKnownNote::from_note(note) {
+            if well_known_note.is_compatible_with(self) {
+                NoteAccountCompatibility::Maybe
+            } else {
+                NoteAccountCompatibility::No
+            }
+        } else {
+            verify_note_script_compatibility(
+                note.script(),
+                component_proc_digests(self.interfaces()),
+            )
+        }
+    }
+}
+
+impl From<&Account> for AccountInterface {
+    fn from(account: &Account) -> Self {
+        let interfaces = AccountComponentInterface::from_procedures(account.code().procedures());
+        let mut auth = Vec::new();
+        interfaces.iter().for_each(|interface| {
+            if let AccountComponentInterface::RpoFalcon512(storage_offset) = interface {
+                auth.push(AuthScheme::RpoFalcon512 {
+                    pub_key: rpo_falcon512::PublicKey::new(
+                        *account
+                            .storage()
+                            .get_item(*storage_offset)
+                            .expect("invalid storage offset of the public key"),
+                    ),
+                })
+            }
+        });
+
+        Self {
+            account_id: account.id(),
+            auth,
+            interfaces,
+        }
+    }
+}
+
+// ACCOUNT COMPONENT INTERFACE
+// ================================================================================================
+
+/// The enum holding all possible account interfaces which could be loaded to some account.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum AccountComponentInterface {
+    /// Exposes procedures from the [`BasicWallet`][crate::account::wallets::BasicWallet] module.
+    BasicWallet,
+    /// Exposes procedures from the
+    /// [`BasicFungibleFaucet`][crate::account::faucets::BasicFungibleFaucet] module.
+    BasicFungibleFaucet,
+    /// Exposes procedures from the
+    /// [`RpoFalcon512`][crate::account::auth::RpoFalcon512] module.
+    ///
+    /// Internal value holds the storage offset where the public key for the RpoFalcon512
```

Comment:
nit: The inner value contains the absolute storage index, right? In that case, instead of "offset" I think I'd just name it as "storage index" to avoid ambiguity. "Offset" makes it sound relative to some other number/index and instead most of the account storage interface deals with absolute "storage indices" so I think it fits slightly better (ie, it is equivalent to the component's storage offset because it only holds one storage value).